mailbox/api/src/main/java/org/apache/james/mailbox/MessageIdManager.java

```diff
@@ -39,4 +42,6 @@
     void delete(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException;
 
     void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException;
+
+    MessageMetaData appendMessage(InputStream msgIn, Date internalDate, Flags flagsToBeSet, MailboxId mailboxId, MailboxSession mailboxSession) throws MailboxException;
```

Comment:
Really ?

I did not wanted to leak the factory for message Id any further, and it seems logic to me that the manager affects an ID to the message ;-)